lib/matplotlib/widgets.py

```diff
@@ -702,14 +702,26 @@ def __init__(
             f'marker{k}': v for k, v in {**defaults, **handle_style}.items()
         }
 
+        verts = np.zeros([5, 2])
+
         if orientation == "vertical":
             self.track = Rectangle(
                 (.25, 0), .5, 2,
                 transform=ax.transAxes,
                 facecolor=track_color
             )
             ax.add_patch(self.track)
-            self.poly = ax.axhspan(valinit[0], valinit[1], 0, 1, **kwargs)
+            verts[0] = .25, valinit[0]
+            verts[1] = .25, valinit[1]
+            verts[2] = .75, valinit[1]
+            verts[3] = .75, valinit[0]
+            verts[4] = .25, valinit[0]
```

Comment:
Since this verts filling code is now used in multiple places (here, and in `set_val`) maybe we should pull it out into a function.